Make warnings show up in CI reports that don't run Toil in-process

adamnovak opened the pull request:
~The warnings disappeared around when I merged https://github.com/vgteam/vg/pull/998, but those commits don't seem to actually be the cause.~

~Now the logger needs to be manually configured for some reason, and the message format is slightly different. But this makes it work again.~

glennhickey commented:
The warnings were lost due to this copypaste-o:

https://github.com/vgteam/vg/pull/998/files#diff-e49ccbeaca6d6c530c3fa11683f0cf05R209

There's a simulfix in #1002



On Wed, Aug 30, 2017 at 1:26 PM, Adam Novak <notifications@github.com>
wrote:

> The warnings disappeared around when I merged #998
> <https://github.com/vgteam/vg/pull/998>, but those commits don't seem to
> actually be the cause.
>
> Now the logger needs to be manually configured for some reason, and the
> message format is slightly different. But this makes it work again.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/vgteam/vg/pull/1003
> Commit Summary
>
>    - Make warnings show up in CI reports again
>
> File Changes
>
>    - *M* jenkins/mine-logs.py
>    <https://github.com/vgteam/vg/pull/1003/files#diff-0> (2)
>    - *M* jenkins/vgci.py
>    <https://github.com/vgteam/vg/pull/1003/files#diff-1> (5)
>
> Patch Links:
>
>    - https://github.com/vgteam/vg/pull/1003.patch
>    - https://github.com/vgteam/vg/pull/1003.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vgteam/vg/pull/1003>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA2_7ih9iBoXmQKNeXqxA34SeYVtuayjks5sdZtBgaJpZM4PHuFm>
> .
>

adamnovak commented:
Hm. Looks like I added some kind of testing line and forgot to remove it.

I think I got even more confused when developing *this* patch; I started with the commit before all the Unicode stuff and added a test warning line at the beginning of one of the short non-mapeval tests. It didn't show up in the output.

I think we've been relying on running Toil in-process in order to get logging set up. We only ever used warnings after doing mapeval, which we do by running a Toil workflow in the current Python, which results in Toil configuring logging for us. That explains why it wasn't set up for the other tests, and why when I run basicConfig I end up with a different format.

So this PR addresses *a* case when warnings won't be displayed, but *not* the case that's actually been giving us trouble (which is fixed in #1002).

Assuming pre-configuring the logger doesn't give us trouble when Toil tries to configure it later, we probably actually want both sets of changes.

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/400/) complete for [PR 1003](https://github.com/vgteam/vg/pull/1003).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/1f4a674460f90f5d116824e4b344ef66e22c6b19/index.html).

9 tests passed, 0 tests failed and 10 tests skipped in 9959 seconds
